src/org/opensolaris/opengrok/history/RCSRepository.java

```diff
@@ -127,7 +128,23 @@ void update() throws IOException {
     @Override
     boolean isRepositoryFor(File file) {
         File rcsDir = new File(file, "RCS");
-        return rcsDir.isDirectory();
+        if (!rcsDir.isDirectory()) {
+            return false;
+        }
+
+        // If there is at least one entry with the ',v' suffix,
+        // consider this a RCS repository.
+        String[] list = rcsDir.list(new FilenameFilter() {
+            @Override
+            public boolean accept(File dir, String name) {
+                // Technically we should check whether the entry is a file
+                // however this would incur additional I/O. The pattern
+                // should be enough.
+                return name.matches(".*,v");
+            }
+        });
+
+        return (list.length > 0);
```

Comment:
yes. JavaDoc says: "If this abstract pathname does not denote a directory, then this method returns null. Otherwise an array of strings is returned, one for each file or directory in the directory."